### PR TITLE
Send mediaHits (location=0) for linear without DVR

### PR DESCRIPTION
**Pull request by chanakabr**
@OrenMe , @itaykinnrot - Can you please review/merge? We need this for Mediacorp Linear concurrency check.

### CODE REVIEW

File: modules/Tvpapi/resources/tvpapiAnalytics.js [hunk 1]
```diff
@@ -127,7 +129,7 @@
 				"mediaType": 0,
 				"iMediaID": this.getProxyConfig('MediaID'),
 				"iFileID": this.fileId,
-				"iLocation": (this.getPlayer().isLive()&&!this.getPlayer().isDVR())?0:this.getCurrentTime()
+				"iLocation": (this.getPlayer().isLive() && !this.getPlayer().isDVR())?0:this.getCurrentTime()
```
Comment (OrenMe) [inline modules/Tvpapi/resources/tvpapiAnalytics.js]: @chanakabr same here, extract the live and dvr to separate params.

File: modules/Tvpapi/resources/tvpapiAnalytics.js [hunk 2]
```diff
@@ -103,10 +103,12 @@
...
 		sendMediaHit: function() {
-			// Do not send media hit in the following conditions: concurrent limit, current time not updated (might be error)
-			// If isLive but not DVR then continue
-			if(this.concurrentFlag || (!(this.getPlayer().isLive()&&!this.getPlayer().isDVR())&&this.getPlayer().getPlayerElementTime() === 0)){
-				return;
+		    var isLive = this.getPlayer().isLive();
+		    var isDvr = this.getPlayer().isDVR();
+
+			// Do not send media hit in the following conditions: concurrent limit, current time not updated (might be error) when media is not (live and not dvr)
+			if(this.concurrentFlag || (!isLive && isDvr && this.getPlayer().getPlayerElementTime() === 0) ){
```
Comment (OrenMe) [inline modules/Tvpapi/resources/tvpapiAnalytics.js]: @chanakabr not sure the condition here is correct - not isLive and is DVR - this condition cannot be met AFAICS

Comment (chanakabr) [reply modules/Tvpapi/resources/tvpapiAnalytics.js]: @OrenMe  -  its ! (isLive && !isDVR) becoming  !isLive && isDVR. Right no?

Comment (chanakabr) [reply modules/Tvpapi/resources/tvpapiAnalytics.js]: @OrenMe - Sorry typo. Fixing

File: modules/Tvpapi/resources/tvpapiAnalytics.js [hunk 3]
```diff
@@ -103,9 +103,12 @@
...
 
 		sendMediaHit: function() {
-			// Do not send media hit in the following conditions: concurrent limit, current time not updated (might be error)
-			if(this.concurrentFlag || this.getPlayer().getPlayerElementTime() === 0 ){
-				return;
+		    var isLive = this.getPlayer().isLive();
+		    var isDvr = this.getPlayer().isDVR();
+
+			// Do not send media hit in the following conditions: concurrent limit, current time not updated (might be error) when media is not (live and not dvr)
+			if(this.concurrentFlag || !(isLive && !isDvr) && this.getPlayer().getPlayerElementTime() === 0) ){
```
Comment (OrenMe) [inline modules/Tvpapi/resources/tvpapiAnalytics.js]: The ! negation should turn && to || so it is 
`!(isLive && !isDvr) ===> (!isLive || isDvr)`

if we already doing a complex expression simpler, let's move `currentTime = this.getPlayer().getPlayerElementTime()` and add it as `currentTime === 0` in the if expression.

File: modules/Tvpapi/resources/tvpapiAnalytics.js [hunk 4]
```diff
@@ -126,7 +129,7 @@
 				"mediaType": 0,
 				"iMediaID": this.getProxyConfig('MediaID'),
 				"iFileID": this.fileId,
-				"iLocation": this.getCurrentTime()
+				"iLocation": (this.getPlayer().isLive() && !this.getPlayer().isDVR())?0:this.getCurrentTime()
```
Comment (OrenMe) [inline modules/Tvpapi/resources/tvpapiAnalytics.js]: Let's move this also to separate variables and use them here to shorten the condition.

File: modules/Tvpapi/resources/tvpapiAnalytics.js (hunk 2 above)
Comment (chanakabr) [reply modules/Tvpapi/resources/tvpapiAnalytics.js]: @OrenMe - Done.

File: modules/Tvpapi/resources/tvpapiAnalytics.js (hunk 1 above)
Comment (chanakabr) [reply modules/Tvpapi/resources/tvpapiAnalytics.js]: @OrenMe - done.

**Comment by chanakabr** [comment]
@OrenMe - Tested analytics below environments.
- HLS live on Chrome and Safari
- HLS (vod) on Safari
- DASH on Chrome

**Comment by OrenMe** [comment]
closing due to inactivity, please reopen to latest if still relevant

**Pull request closed without merging**